src/state/reducer.ts

```diff
@@ -1,8 +1,11 @@
 import { act } from 'react-dom/test-utils';
-import { State } from '../types';
+import { State, Status } from '../types';
 import { TimerActionsType } from './actions';
 
 function reducer(state: State, action: TimerActionsType) {
+  const paused:Status = 'PAUSED';
+  const running:Status = 'RUNNING';
+  const stopped:Status = 'STOPPED';
```

Comment:
If fact, we can just tell that our function returns type `State` (it was missing ;-)). That way, inside each switch case, TypeScript knows that `status` property is of type `Status`: no need to create typed variables.

```suggestion
function reducer(state: State, action: TimerActionsType): State {
```

Reply:
Thanks for this! I was having a hard time trying to figure this out. My initial commit was failing because it wasn't recognizing the new type `status`. It thought it was a `string`.